src/NuGet.Core/NuGet.CommandLine.XPlat/UILanguageOverride.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+
+namespace NuGet.CommandLine.XPlat
+{
+    /// <summary>
+    /// Copied from https://github.com/dotnet/sdk/blob/49d9b4148c4f65fd3f691186a4533375c3a83c97/src/Cli/dotnet/UILanguageOverride.cs#L9
+    /// </summary>
+    internal static class UILanguageOverride
+    {
+        private const string DOTNET_CLI_UI_LANGUAGE = nameof(DOTNET_CLI_UI_LANGUAGE);
+        private const string VSLANG = nameof(VSLANG);
+        private const string PreferredUILang = nameof(PreferredUILang);
+
+        public static void Setup()
+        {
+            CultureInfo language = GetOverriddenUILanguage();
+            if (language != null)
+            {
+                ApplyOverrideToCurrentProcess(language);
+                FlowOverrideToChildProcesses(language);
+            }
+        }
+
+        private static void ApplyOverrideToCurrentProcess(CultureInfo language)
+        {
+            CultureInfo.DefaultThreadCurrentUICulture = language;
+        }
+
+        private static void FlowOverrideToChildProcesses(CultureInfo language)
+        {
+            // Do not override any environment variables that are already set as we do not want to clobber a more granular setting with our global setting.
+            SetIfNotAlreadySet(DOTNET_CLI_UI_LANGUAGE, language.Name);
+            SetIfNotAlreadySet(VSLANG, language.LCID); // for tools following VS guidelines to just work in CLI
+            SetIfNotAlreadySet(PreferredUILang, language.Name); // for C#/VB targets that pass $(PreferredUILang) to compiler
+        }
+
+        private static CultureInfo GetOverriddenUILanguage()
+        {
+            // DOTNET_CLI_UI_LANGUAGE=<culture name> is the main way for users to customize the CLI's UI language.
+            string dotnetCliLanguage = Environment.GetEnvironmentVariable(DOTNET_CLI_UI_LANGUAGE);
+            if (dotnetCliLanguage != null)
+            {
+                try
+                {
+                    return new CultureInfo(dotnetCliLanguage);
+                }
+                catch (CultureNotFoundException) { }
```

Comment:
Since this is an environment variable that a user may type in manually, its probably worth telling them that what they specified isn't valid right?

Reply:
That sounds like a good idea. Actually, NuGet has a `--force-english-output` flag in some commands, such as `dotnet nuget locals`. Would it make sense to show a message at 'detail' level?

Reply:
We should probably have the ENV VAR name in the message as well.